diff --git a/pyproject.toml b/pyproject.toml
index 9b75fcf35..d9c62ecdd 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -74,7 +74,7 @@ mlm = [
 
 dev = [
     "nvidia-modelopt[torch]; sys_platform != 'darwin'",
-    "transformer-engine[pytorch,core_cu13]>=2.9.0a0,<2.12.0",
+    "transformer-engine[pytorch,core_cu13]>=2.9.0a0,<2.13.0",
     "nvidia-resiliency-ext",
     "tqdm",
     "einops~=0.8",
